numeric.h

```diff
@@ -0,0 +1,34 @@
+#ifndef NUMBER_THEORY_NUMERIC_H
+#define NUMBER_THEORY_NUMERIC_H
+
+#include <limits>
+#include <numeric>
+#include <utility>
+
+/**
+ * Common numeric functions.
+ */
+
+namespace number_theory {
+
+using std::gcd;
+using std::lcm;
+
+/**
+ * Extended Euclidean algorithm.
+ * Given two numbers a and b, return a pair (x, y)
+ * satisfying x*a + y*b == gcd(a,b).
+ */
+template <class T,
+          std::enable_if_t<std::numeric_limits<T>::is_signed, bool> = true>
+std::pair<T, T> exgcd(const T &a, const T &b) {
+  if (b == 0)
+    return std::make_pair(1, 0);
+  auto [y, x] = exgcd(b, a % b);
```

Comment:
What about using iterative implementations over recursive ones in general?

Reply:
Unlike the common gcd function, exgcd is a non-tail recursive function, so it is hard to have an iterative implementation.

Reply:
I found [this](https://cp-algorithms.com/algebra/extended-euclid-algorithm.html) implementation. Maybe we can use their implementation but we should be careful about its license.

Reply:
done